File: src/main/java/com/temelyan/pomoapp/repository/UserRepopsitory.java
package com.temelyan.pomoapp.repository;

import com.temelyan.pomoapp.model.User;

import java.time.LocalDateTime;

public interface UserRepopsitory {
    User save(User user);

    User get(int id);

    User getWithProjects(int id);

    User getByEmail(String email);

    User findUserByResetToken(String token);

    User getByIdWithPomosInDateRange(int userId, LocalDateTime from, LocalDateTime to);
}

File: src/main/java/com/temelyan/pomoapp/repository/UserRepopsitory.kt
package com.temelyan.pomoapp.repository

import com.temelyan.pomoapp.model.User

interface UserRepopsitory {
    fun save(user: User): User

    fun get(id: Int): User

    fun getByEmail(email: String): User

    fun findUserByResetToken(token: String): User
}
